src/spikeinterface/preprocessing/tests/test_scaling.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+import numpy as np
+from spikeinterface.core.testing_tools import generate_recording
+from spikeinterface.preprocessing import ScaleTouVRecording  # Replace 'your_module' with your actual module name
+
+
+def test_scale_to_uv():
+    # Create a sample recording extractor with fake gains and offsets
+    num_channels = 4
+    sampling_frequency = 30_000.0
+    durations = [1]  # seconds
+    recording = generate_recording(
+        num_channels=num_channels,
+        durations=durations,
+        sampling_frequency=sampling_frequency,
+    )
+
+    rng = np.random.default_rng(0)
+    gains = rng.random(size=(num_channels)).astype(np.float32)
+    offsets = rng.random(size=(num_channels)).astype(np.float32)
+    recording.set_channel_gains(gains)
+    recording.set_channel_offsets(offsets)
+
+    # Apply the preprocessor
+    scaled_recording = ScaleTouVRecording(recording=recording)
+
+    # Check if the traces are indeed scaled
+    expected_traces = recording.get_traces(return_scaled=True)
```

Comment:
what happens if `return_scaled` is `False` on the `ScaleTouVRecording?` intuitively I might expect it always returns scaled. Though of course it makes more sense for it only to return scaled to uV when `return_scaled=True`. Maybe this could be added to docstring? and a quick additional test that the `return_scaled=False` case does not change?

Reply:
This is a good idea and something that I did not think before.

Reply:
Shouldn't we raise an error instead. scaling to uv necessarily requires us to return_scaled. If the user is not okay with that, then they shouldn't use this class? Why should we let them @JoeZiminski ? What use-case?

Reply:
Mmm the thing is that return_scaled is False by default. Having to set `return_scaled=True` even when passing this class defeats the purpose of using it I feel.

Reply:
Oh yes, if `return_scaled=False` had to be set explicitly I would agree Zach in this case though I guess we let it slide as it is the default. 

If I understand everything correctly, maybe a test can just check the output of `return_scaled=True` and `return_scaled=False` are identical?

Reply:
What I meant above when I have to think more is that the real problem is if someone calls `return_scaled=True` then the scaling would be done twice. I need to think about this.

Reply:
This is critical in cases of things like the sorting_analyzer (which I think is unique?) that have `return_scaled=True`.

Reply:
Interesting yes, in this cause should the gain and offset be set to 1 and 0? Is there any other workaround except `return_scaled=True` raising an error? Would there be any issue in changing these apart from lost provenance about the original recording? I think it is 'accurate' in some sense (the gain and offset to uV is really now 1 and 0 after `scale_to_uV`).

Reply:
Thanks for pushing on this. I think that yes, the gains and offsets are propagated in preprocessing chains here:

https://github.com/h-mayorquin/spikeinterface/blob/a98e81af11f06c0ee47338b42e3132964f558d76/src/spikeinterface/preprocessing/basepreprocessor.py#L25-L26

I think we should modify the gains and offsets so any step down the chain should get 1 and 0s as gains and offsets respectively. The provenance is not lost because the dict still keeps the properties at the parent levels. Let me think abit more about it.

Reply:
OK, I added this, the provenance is mantained because the gains and offsets are copied from parent on further forward on the chain which is exactly what we want. I added a test that guarantess that return_scaled=True has no effect once gain_to_uV is part of the preprocessing chain.